### PR TITLE
Compute packages available in analyzed Python package release

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Related: https://github.com/thoth-station/thoth-application/issues/1280

## This introduces a breaking change

- [x] No

### CODE REVIEW

**Comment by fridex** [comment]
Examples:

**thoth-common==0.29.1** (namespaced package)

```json
        "packages": [
          "thoth.common",
           "thoth.common.config"
        ],
```

**scikit-learn** (package name not matching)

```json
        "packages": [
          "sklearn",
          "sklearn.__check_build",
          "sklearn._build_utils",
          "sklearn._loss",
          "sklearn.cluster",
          "sklearn.compose",
          "sklearn.covariance",
          "sklearn.cross_decomposition",
          "sklearn.datasets",
          "sklearn.decomposition",
          "sklearn.ensemble",
          "sklearn.experimental",
          "sklearn.externals",
          "sklearn.feature_extraction",
          "sklearn.feature_selection",
          "sklearn.gaussian_process",
          "sklearn.impute",
          "sklearn.inspection",
          "sklearn.linear_model",
          "sklearn.manifold",
          "sklearn.metrics",
          "sklearn.mixture",
          "sklearn.model_selection",
          "sklearn.neighbors",
          "sklearn.neural_network",
          "sklearn.preprocessing",
          "sklearn.semi_supervised",
          "sklearn.svm",
          "sklearn.tests",
          "sklearn.tree",
          "sklearn.utils",
          "sklearn._loss.tests",
          "sklearn.cluster.tests",
          "sklearn.compose.tests",
          "sklearn.covariance.tests",
          "sklearn.cross_decomposition.tests",
          "sklearn.datasets.tests",
          "sklearn.decomposition.tests",
          "sklearn.ensemble._hist_gradient_boosting",
          "sklearn.ensemble.tests",
          "sklearn.experimental.tests",
          "sklearn.feature_extraction.tests",
          "sklearn.feature_selection.tests",
          "sklearn.gaussian_process.tests",
          "sklearn.impute.tests",
          "sklearn.inspection._plot",
          "sklearn.inspection.tests",
          "sklearn.linear_model._glm",
          "sklearn.linear_model.tests",
          "sklearn.manifold.tests",
          "sklearn.metrics._plot",
          "sklearn.metrics.cluster",
          "sklearn.metrics.tests",
          "sklearn.mixture.tests",
          "sklearn.model_selection.tests",
          "sklearn.neighbors.tests",
          "sklearn.neural_network.tests",
          "sklearn.preprocessing.tests",
          "sklearn.semi_supervised.tests",
          "sklearn.svm.tests",
          "sklearn.tree.tests",
          "sklearn.utils.tests",
          "sklearn.ensemble._hist_gradient_boosting.tests",
          "sklearn.inspection._plot.tests",
          "sklearn.linear_model._glm.tests",
          "sklearn.metrics._plot.tests",
          "sklearn.metrics.cluster.tests"
        ],
````

**Comment by fridex** [comment]
/hold

... will add tests, some of them will need to be adjusted

**Review by KPostOffice — COMMENTED** [review]
/lgtm

File: thoth/solver/python/python.py [hunk 1]
```diff
@@ -417,6 +417,17 @@ def resolve(requirements, *, index_urls, python_version, exclude_packages, trans
         result["unparsed"].extend(solver_result["unparsed"])  # type: ignore
         result["unresolved"].extend(solver_result["unresolved"])  # type: ignore
 
+    for item in result["tree"]:
+        packages = []
+        for file_info in item["importlib_metadata"]["files"]:
```
Comment (goern) [inline thoth/solver/python/python.py:422]: `item["importlib_metadata"]` could be None, should that be checked?

**Pull request merged**